File: 104. Maximum Depth of Binary Tree/maxDepth.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        // empty
        if (root == NULL) { return 0; }
        // BFS
        int depth = 0;
        queue<TreeNode*> nodeList;
        nodeList.push(root);

        while (!nodeList.empty()) {
            // same depth
            int n = nodeList.size();
            for(int i=0; i<n; i++){
                // popup(0)
                TreeNode* node = nodeList.front();
                nodeList.pop();
                // push_back
                if(node->left != NULL) { nodeList.push(node->left); }
                if(node->right != NULL) { nodeList.push(node->right); }
            }
            // next depth
            depth++;
        }
        return depth;
    }
};

File: 104. Maximum Depth of Binary Tree/maxDepth.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {
    if (root) {
        return Math.max(maxDepth(root.left), maxDepth(root.right))+1
    }
    else {
        return 0
    }    
};

// Runtime 76 ms / Memory 44.7 MB
// debug: https://jsfiddle.net/